test/pkg/environment/azure/rbac_manager.go

```diff
@@ -0,0 +1,106 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization/v2"
+	"github.com/google/uuid"
+)
+
+type RBACManager struct {
+	subscriptionID string
+	client         *armauthorization.RoleAssignmentsClient
+}
+
+// NewRBACManager builds a client with DefaultAzureCredential.
+func NewRBACManager(subscriptionID string) (*RBACManager, error) {
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return nil, err
+	}
+	c, err := armauthorization.NewRoleAssignmentsClient(subscriptionID, cred, &arm.ClientOptions{
```

Comment:
This retry configuration is duplicated in at least 2 places (maybe 3 if `azkeys.ClientOptions` is an alias). Share it (i.e. put it in a function that constructs it and re-use the function)

Reply:
azkeys.ClientOptions isn't an alias to arm.ClientOptions :( its an alias for azcore.ClientOptions. 

Made a reusable function for other cases, removed the case for rbac manager, it probably doesn't need to retry assignment, it shouldn't get 403 on assignment anymore since we removed some of the retry stuff from the clientside princpal retries and added principalType.